lib/puppet/provider/exec/powershell.rb

```diff
@@ -27,28 +27,30 @@
         }
   EOT
 
-  POWERSHELL_MODULE_UPGRADE_MSG = <<-UPGRADE
-  Currently, the PowerShell module has reduced v1 functionality on this agent
-  due to one or more of the following conditions:
+  unless defined?(POWERSHELL_MODULE_UPGRADE_MSG)
+    POWERSHELL_MODULE_UPGRADE_MSG = <<-UPGRADE
+    Currently, the PowerShell module has reduced v1 functionality on this agent
+    due to one or more of the following conditions:
 
-  - Puppet 3.x (non-x64 version)
+    - Puppet 3.x (non-x64 version)
 
-    Puppet 3.x uses a Ruby version that requires a library to support a colored
-    console. Unfortunately this library prevents the PowerShell module from
-    using a shared PowerShell process to dramatically improve the performance of
-    resource application.
+      Puppet 3.x uses a Ruby version that requires a library to support a colored
+      console. Unfortunately this library prevents the PowerShell module from
+      using a shared PowerShell process to dramatically improve the performance of
+      resource application.
 
-  - PowerShell v2 with .NET Framework 2.0
+    - PowerShell v2 with .NET Framework 2.0
 
-    PowerShell v2 works with both .NET Framework 2.0 and .NET Framework 3.5.
-    To be able to use the enhancements, we require .NET Framework 3.5.
-    Typically you will only see this on a base Windows Server 2008 (and R2)
-    install.
+      PowerShell v2 works with both .NET Framework 2.0 and .NET Framework 3.5.
+      To be able to use the enhancements, we require at least .NET Framework 3.5.
```

Comment:
`at least` was removed in commit https://github.com/puppetlabs/puppetlabs-powershell/commit/18628d6d4dc690dfe52430f53ebbfc2dbdc02e95. This appears to be revertting that.